modules/idle-scheduler/src/lib.rs

```diff
@@ -76,14 +86,53 @@ pub mod module {
 	#[pallet::getter(fn next_task_id)]
 	pub type NextTaskId<T: Config> = StorageValue<_, Nonce, ValueQuery>;
 
+	///
+	#[pallet::storage]
+	#[pallet::getter(fn previous_relay_block)]
+	pub type PreviousRelayBlockNumber<T: Config> = StorageValue<_, BlockNumber, ValueQuery>;
+
 	#[pallet::pallet]
 	#[pallet::without_storage_info]
 	pub struct Pallet<T>(_);
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			// Unwrap shouldn't fail if it does defaults to 0
+			// This is the previous relay block because `on_initialize` is executed before
+			// the inherent that sets the new relay chain block number
+			let previous_relay_block: BlockNumber = T::RelayChainBlockNumberProvider::current_block_number()
+				.try_into()
+				.unwrap_or_default();
+			PreviousRelayBlockNumber::<T>::put(previous_relay_block);
+			// One write and one read
+			T::DbWeight::get().reads_writes(One::one(), One::one())
```

Comment:
Should use benchmark. Weight could change if `on_initialize` updated in the future, and it's easy to miss weight update then.